api/src/main/java/org/apache/iceberg/data/Record.java

```diff
@@ -32,7 +32,7 @@
 
   Object get(int pos);
 
-  Record copy();
+  Record copyRecord();
```

Comment:
Both `Record` API and `Manifest` API define the copy method.

Reply:
I don't think this should break the `Record` API. Why can't the class implement both in the same method by returning `GenericManifestFile`?

Reply:
If two interfaces contain a method with the same signature but different return types, then it is impossible to implement both the interface simultaneously.

Reply:
This works for me:

```java
  interface A {
    A copy();
  }
  
  interface B {
    B copy();
  }
  
  public class AB implements A, B {
    @Override
    public AB copy() {
      return this;
    }
  }
```

Reply:
Just learned!